frontend/scripts/react-components/shared/world-map/world-map.component.jsx

```diff
@@ -44,13 +44,18 @@ class WorldMap extends React.PureComponent {
     this.renderCountriesAnnotations = this.renderCountriesAnnotations.bind(this);
   }
 
-  componentDidUpdate(prevProps) {
+  shouldComponentUpdate(nextProps) {
```

Comment:
I think `shouldComponentUpdate` should not have side effects like loading the data. Without introducing extra property into the state like `isLoading` we could avoid this flickering in a similar way.

```
  componentWillReceiveProps(nextProps) {
    if (
      nextProps.selectedContext !== this.props.selectedContext ||
      nextProps.selectedYears !== this.props.selectedYears
    ) {
      this.props.getTopNodes();
    }
  }

  shouldComponentUpdate(nextProps) {
    // data is probably being loaded, do not update component yet
    if (nextProps.flows.length === 0 && this.props.flows.length > 0) {
      return false;
    }

    return true;
  }

```

Reply:
I agree side-effects on `shouldComponentUpdate` is a no go.